node/examples/repl_server.js

```diff
@@ -89,19 +91,24 @@ function ReplSession(repler, options) {
     self.logger = options.logger;
 }
 
+util.inherits(ReplSession, EventEmitter);
+
 ReplSession.prototype.start = function start(stream) {
-    // function startReplSession() {
-    //     setImmediate(startReplSession);
-    // }
     var self = this;
-    self.repler.open(stream);
+    self.once('resize', function onResize(size) {
+        self.repler.open(stream, {
+            width: size.cols,
+            height: size.rows
+        });
+    });
 };
 
 ReplSession.prototype.control = function control(stream) {
     var self = this;
-    stream.on('data', function(obj) {
+    stream.on('data', function onControlData(obj) {
         if (obj.op === 'resize') {
             self.logger.info('resize to', obj);
+            self.emit('resize', obj);
```

Comment:
I don't quite agree with the overly-loose coupling here, can we just have some `self._startReplr' utility method that gets called directly?

To be fair, I'm coming off a round of performance tuning whence EventEmitter (especially .once) showed to be abundantly slow, and this isn't a hot path... so I can live with this if you prefer.